src/SourceLocator/Type/DirectorySourceLocator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: HP
+ * Date: 09/24/2016
+ * Time: 5:29 AM
+ */
+
+namespace BetterReflection\SourceLocator\Type;
+
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\Reflector\Reflector;
+
+class DirectorySourceLocator implements SourceLocator
+{
+
+    /**
+     * @var AggregateSourceLocator
+     */
+    private $aggregatedSourceLocator;
+
+
+    /**
+     * DirectorySourceLocator constructor.
+     * @param $directory string|array directory to scan
+     */
+    public function __construct($directory = null)
```

Comment:
Given that we can build a `DirectorySourceLocator` from a single path, a set of paths or a `RecursiveDirectoryIterator`, would it make sense to move this to a named constructor, while `__construct()` is made `private`?